src/js/types.ts

```diff
@@ -241,41 +241,27 @@ export interface FSType {
   ) => FSNode) & {
     major: number;
   };
-  closeStream: (fd: number) => void;
-  open: (path: string, flags: string | number, mode?: number) => FSStream;
-  makedev: (major: number, minor: number) => number;
-  mkdev: (path: string, dev: number) => FSNode;
-  filesystems: any;
-  stat: (path: string, dontFollow?: boolean) => any;
-  readdir: (path: string) => string[];
-  isDir: (mode: number) => boolean;
-  isMountpoint: (mode: FSNode) => boolean;
   lookupPath: (
     path: string,
     options?: {
       follow_mount?: boolean;
     },
   ) => { node: FSNode };
-  isFile: (mode: number) => boolean;
-  writeFile: (path: string, contents: any, o?: { canOwn?: boolean }) => void;
-  chmod: (path: string, mode: number) => void;
-  utime: (path: string, atime: number, mtime: number) => void;
-  rmdir: (path: string) => void;
-  mount: (type: any, opts: any, mountpoint: string) => any;
-  unmount: (mountpoint: string) => any;
-  write: (
-    stream: FSStream,
-    buffer: any,
-    offset: number,
-    length: number,
-    position?: number,
-  ) => number;
-  close: (stream: FSStream) => void;
-  ErrnoError: { new (errno: number): Error };
+  open: (path: string, flags: string | number, mode?: number) => FSStream;
+  filesystems: any;
+  isMountpoint: (node: FSNode) => boolean;
+  closeStream: (fd: number) => void;
   registerDevice<T>(dev: number, ops: FSStreamOpsGen<T>): void;
-  syncfs(dir: boolean, oncomplete: (val: void) => void): void;
+  writeFile: (path: string, contents: any, o?: { canOwn?: boolean }) => void;
 }
 
+/**
+ * Combined filesystem type that omits the incompatible lookupPath from `@types/emscripten` and adds Pyodide-specific filesystem methods.
+ * TODO: Consider upstreaming these APIs to `@types/emscripten`
+ * @hidden
+ */
+export type FSType = Omit<typeof FS, "lookupPath"> & PyodideFSType;
```

Comment:
sphinx-js fails to handle this type, but as this object is hidden in the docs, I think the warning can be ignored.

```
<unknown>:1: WARNING: js:class reference target not found: Omit [ref.class]
<unknown>:1: WARNING: js:class reference target not found: FS [ref.class]
<unknown>:1: WARNING: js:class reference target not found: Omit [ref.class]
```

@hoodmane Do you know a workaround for this? We have enabled `fail_on_warning` on RTD, to the doc build is failing in CI.

Reply:
Okay, it looks like adding the following line to [`mdn_xrefs.py`](https://github.com/pyodide/pyodide/blob/38c4dac21efbdf9407b23e03550792f773242ad6/docs/sphinx_pyodide/sphinx_pyodide/mdn_xrefs.py#L181-L185) will do the job (we can probably upstream it to sphinx-js though).

```
    (
        "js:class",
        "Omit",
        "https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys",
    ),
    (
        "js:class",
        "FS",
        "https://emscripten.org/docs/api_reference/Filesystem-API.html",
    )
```

@R3gardless could you update the file that I linked accordingly, to prevent the parser error in docs?

Reply:
@ryanking13 Thanks for the feedback

I updated the corresponding files.